src/yugabyte-bash-common.sh

```diff
@@ -59,18 +59,19 @@ yb_virtualenv_basename=venv
 # Bash version validation
 # -------------------------------------------------------------------------------------------------
 
-if [[ "$BASH_VERSION" =~ ^3[.] ]]; then
-  if "$is_mac"; then
-    log "Bash 3 detected, using Homebrew to install Bash 4 or later if not already installed"
-    homebrew_bash_path=/usr/local/bin/bash
-    if [[ ! -f $homebrew_bash_path ]]; then
-      # We can't use "set -x" in a subshell in Bash 3, because the error code is not propagated.
-      ( set -x; brew install bash )
+check_mac_bash_version() {
+  if [[ "$BASH_VERSION" =~ ^3[.] ]]; then
+    if "$is_mac"; then
+      log "Bash 3 detected, using Homebrew to install Bash 4 or later if not already installed"
+      homebrew_bash_path=/usr/local/bin/bash
       if [[ ! -f $homebrew_bash_path ]]; then
```

Comment:
This is a common functions file. If we are in the wrong shell, I think we should give an error and exit out, not install software.